odoo/addons/base/models/res_lang.py

```diff
@@ -279,13 +281,20 @@ def toggle_active(self):
             mods = self.env['ir.module.module'].search([('state', '=', 'installed')])
             mods._update_translations(active_lang)
 
-    @api.model_create_multi
-    def create(self, vals_list):
-        self.clear_caches()
-        for vals in vals_list:
-            if not vals.get('url_code'):
-                vals['url_code'] = vals.get('iso_code') or vals['code']
-        return super(Lang, self).create(vals_list)
+        # If we activate a lang, set it's url_code to the shortest version if possible
+        if self.active and len(self.url_code) > 3:   # mya is the shortest
+            short = self.code.split('_')[0]
+            others = self.with_context(active_test=False).search([('active', '=', False), ('url_code', '=', short)])
+            if others and others != self:
+                others.url_code = others.code
+            if not self.search([('url_code', '=', short)]):
+                self.url_code = short
```

Comment:
Cool :+1: 
But I would not to that in `toggle_active` but in `write` IMHO, both will end up calling `write` but we can't be sure all the active switch are going through the `toggle_active`.

Reply:
I did that initially, but reverted it back; it's easier to not overwrite write to avoid unpredictable side effects. (especially for tests). At least you can write whatever you want in write, it will not force a short version.

Reply:
Fair :+1: , I don't really mind, it's a very advanced feature of a very technical record, let's say it's a bonus when enabling using the toggle_active